server/models/PetType.js

```diff
@@ -0,0 +1,87 @@
+import pg from "pg"
+
+const pool = new pg.Pool({ connectionString: "postgres://postgres:password@localhost:5432/new_family_member_development" })
+
+class PetType {
+    constructor({ id, type, img_url, imgUrl, description }) {
+        this.id = id
+        this.type = type
+        this.imgUrl = imgUrl || img_url
+        this.description = description
+    }
+    static async findAll() {
+        try {
+            const result = await pool.query("SELECT * FROM pet_types;")
+            const typesData = result.rows
+            const types = typesData.map(type => new this(type))
+
+            return types
+
+        } catch (error) {
+            console.log(error)
+            throw (error)
+        }
+    }
+
+    static async findByType(type) {
+        const petFile = await import('./Pet.js')
+        const Pet = petFile.default
+        let petTypeId
+        if (type === "cats") {
+            petTypeId = 1
+        } if (type === "dogs") {
+            petTypeId = 2
+        }
+        try {
+            const query = "SELECT * FROM adoptable_pets WHERE pet_type_id = $1;"
+            const result = await pool.query(query, [petTypeId])
+            const petsData = result.rows
+            const pets = petsData.map(pet => new Pet(pet))
+            return pets
+        } catch (error) {
+            console.log(error)
+            throw (error)
+        }
+    }
+
+    static async findById(Id) {
+    }
+
+    // static async findByType(type) {
+
+    //     try {
+    //         const query = "SELECT * FROM pet_types WHERE type = $1;"
+    //         const result = await pool.query(query, [type])
+
+    //         const typesData = result.rows[0]
+    //         const petType = new this(typesData)
+
+    //         return petType
+
+    //     } catch (error) {
+    //         console.log(error)
+    //         throw (error)
+    //     }
+    // }
+
+    // async pets() {
+    //     const petFile = await import("./Pet.js")
+    //     const Pet = petFile.default
+
+    //     try {
+    //       const query = `SELECT * FROM adoptable_pets WHERE pet_type_id = $1;`
+    //       const result = await pool.query(query, [this.id])
+
+    //       const relatedPetsData = result.rows
+    //       const relatedPets = relatedPetsData.map(pet => new Pet(pet))
+
+    //       return relatedPets
+
+    //     } catch (error) {
+    //         console.log(error)
+    //         throw (error)
+    //     }
+    // }
```

Comment:
We can either comment these methods in if they are to be used, otherwise we should remove them and add them in when they are needed for a feature - looks like your Router is only using the first 2 methods, in which case these unused methods should be removed for this current feature!

Reply:
OK, all commented out code has been deleted!